src/Admin/FieldDescriptionInterface.php

```diff
@@ -15,6 +15,8 @@
 
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @method string|null getTargetModel()
  */
 interface FieldDescriptionInterface
```

Comment:
I'm moving this class in the DatagridBundle, in order to reduce the duplicate classes between the AdminBundle and the DatagridBundle.

Can you take a look at https://github.com/sonata-project/SonataDatagridBundle/pull/234 ?

Reply:
That decision seems right. Your proposal is to implement the model manager specific methods at each [model manager](https://github.com/sonata-project/SonataDatagridBundle/pull/234/commits/1423ac793db96abde04df9bfda8236a1550ba0ae#diff-c660d034b5e324dc2190e06007801451R18)?
If we should wait for that movement first, just let me know if you need some changes here.

Reply:
I don't really know what to do with these methods.

I first thought that I could implement them is a way which works for every model manager.
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Admin/FieldDescription.php
https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/3.x/src/Admin/FieldDescription.php
https://github.com/sonata-project/SonataDoctrinePhpcrAdminBundle/blob/2.x/src/Admin/FieldDescription.php

Are really really similar.
`getTargetEntity` is the only method with a different implementation, and only for Phpcr.

But maybe it's safer like this since we don't know how the model manager will evolve.

What I'm gonna do is
- Add the FieldDescriptionInterface in DatagridBundle 3.x
- Merge into master the FieldDescriptionInterface.
- Update DatagridBundle interface in master branch to be similar to DatagridInterface/PagerInterface from SonataAdminBundle
- Require DatagridBungle by AdminBundle in master branch
- Update AdminBundle code to use DatagridBundle classes

I just recommend that If you want to change DatagridInterface, PagerInterface or FieldDescriptionInterface" to do it now (or to do it later on the DatagridBundle) because if I've reach the last step of my checklist these class will mainly be removed on master for the AdminBundle and this could lead to conflict or a lost of your changes.

Reply:
Don't worry. The only modification related between these deprecations and your changes is this addition I think (`FieldDescriptionInterface::getTargetModel()`). I'll ask you later if some conflict appears while merging these changes into `master`.